pixeltable/func/function.py

```diff
@@ -64,37 +73,114 @@ def display_name(self) -> str:
             return self.self_path[len(ptf_prefix):]
         return self.self_path
 
+    @property
+    def is_polymorphic(self) -> bool:
+        return len(self.signatures) > 1
+
+    @property
+    def signature(self) -> Signature:
+        assert not self.is_polymorphic
+        return self.signatures[0]
+
     @property
     def arity(self) -> int:
+        assert not self.is_polymorphic
         return len(self.signature.parameters)
 
+    def _overload_resolution(self, signature_idx: int) -> Function:
+        """
+        Return a new  `Function` that retains just the single signature at index `signature_idx`, and is otherwise
+        identical to this `Function`.
+        """
+        assert 0 <= signature_idx < len(self.signatures)
+
+        # We cache the resolutions in self.__resolutions. This ensures that each resolution is represented globally
```

Comment:
move this to the class comment, it's a global invariant.

Reply:
Done.
<!-- Reviewable comment -OEWnhq1-UZcdoqkCcaq:b-896fix -->
<!-- Sent from Reviewable.io -->